 Also load legacy provider for blowfish.
Author: Sebastian Ramacher <sramacher@debian.org>
Last-Update: 2022-05-21

--- a/configure.ac
+++ b/configure.ac
@@ -462,10 +462,6 @@
    #endif
    ]])],[AC_MSG_RESULT(OK)],[AC_MSG_ERROR(This release of CFEngine requires OpenSSL >= 0.9.7)])
 
-   if test "x$ac_cv_lib_crypto_RSA_generate_key_ex" = "xyes" ; then
-      AC_DEFINE(OPENSSL_NO_DEPRECATED, 1, [Define if non deprecated API is available.])
-   fi
-
    if test "x$ac_cv_lib_crypto_RSA_generate_key_ex" = "xno" && \
       test "x$ac_cv_lib_crypto_RSA_generate_key" = "xno" ; then
       AC_MSG_ERROR(Cannot find OpenSSL)
--- a/libpromises/crypto.c
+++ b/libpromises/crypto.c
@@ -27,6 +27,7 @@
 #include <openssl/err.h>                                        /* ERR_* */
 #include <openssl/rand.h>                                       /* RAND_* */
 #include <openssl/bn.h>                                         /* BN_* */
+#include <openssl/provider.h>
 #include <libcrypto-compat.h>
 
 #include <cf3.defs.h>
@@ -60,6 +61,8 @@
 /* TODO move crypto.[ch] to libutils. Will need to remove all manipulation of
  * lastseen db. */
 
+static OSSL_PROVIDER* legacy_provider = NULL;
+static OSSL_PROVIDER* default_provider = NULL;
 static bool crypto_initialized = false; /* GLOBAL_X */
 
 const char *CryptoLastErrorString()
@@ -75,6 +78,8 @@
         SetupOpenSSLThreadLocks();
         OpenSSL_add_all_algorithms();
         OpenSSL_add_all_digests();
+        legacy_provider = OSSL_PROVIDER_load(NULL, "legacy");
+        default_provider = OSSL_PROVIDER_load(NULL, "default");
         ERR_load_crypto_strings();
 
         RandomSeed();
@@ -103,6 +108,10 @@
         }
 
         chmod(randfile, 0600);
+
+        OSSL_PROVIDER_unload(legacy_provider);
+        OSSL_PROVIDER_unload(default_provider);
+
         EVP_cleanup();
         CleanupOpenSSLThreadLocks();
         ERR_free_strings();

